core/domain/feedback_services.py

```diff
@@ -366,6 +366,34 @@ def _get_thread_from_model(thread_model):
         thread_model.created_on, thread_model.last_updated)
 
 
+def get_messages_for_exp_id(exp_id, limit=None):
```

Comment:
`exploration_id` isn't a property of a FeedbackMessage object. Such an object is represented by it's thread_id. Thus, I don't think directly querying the messages using exploration_id is possible.

As a simple fix, we could limit the no. of messages fetched through the `get_messages` query to `feconf.NEW_FEEDBACK_COUNT_DASHBOARD` instead of the default query limit:

```diff
@@ -198,7 +198,7 @@ class FeedbackMessageModel(base_models.BaseModel):
                 exploration_id, thread_id))

         return cls.get_all().filter(
-            cls.thread_id == full_thread_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
+            cls.thread_id == full_thread_id).fetch(feconf.NEW_FEEDBACK_COUNT_DASHBOARD)
```